### PR TITLE
NAS-122855 / 23.10 / Refactoring network interface table

### PR DESCRIPTION
**Pull request by undsoft**
Testing: Network -> Interfaces table.

### CODE REVIEW

**Review by denysbutenko — COMMENTED** [review]
Tested on M40. Users can not create or edit network interfaces on HA systems due to enabled failover. Perhaps, it is worth disabling those buttons and adding a tooltip that says, "needs to disable failover to create or edit network interfaces".

The "Edit" button lacks a tooltip.

**Review by denysbutenko — COMMENTED** [review]
The table data reloads when closing Add/Edit form without changes. The expectation is reloading when submitted only.

**Review by denysbutenko — COMMENTED** [review]
image

Two IP addresses make a row nearly twice bigger. Perhaps reducing `margin` or `line-height` could help here.

**Comment by undsoft** [comment]
> Users can not create or edit network interfaces on HA systems due to enabled failover. 

This was the intent. Unfortunately, there is something wrong with how we detect that HA is active. On our m40, because HA is broken, it is not detected. I would suggest mocking `[isHaEnabled]`.
Add and delete buttons will be disabled. Edit button will not be disabled, because I think users may want to view current values in the form.

Rest of the comments addressed.

File: src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-date/ix-cell-date.component.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@ import { Column, ColumnComponent } from 'app/modules/ix-table2/interfaces/table-
   templateUrl: './ix-cell-date.component.html',
 })
 export class IxCellDateComponent<T> extends ColumnComponent<T> {
-  formatDate = 'yyyy-MM-dd';
-  formatTime = 'HH:mm:ss';
+  formatDate: string = undefined;
```
[the remark this answers is not in the document]
Comment (undsoft) [reply src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-date/ix-cell-date.component.ts:8]: I kept these for the possibility of overwriting default values.

**Review by denysbutenko — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**